commands/payment_channel_daemon_test.go

```diff
@@ -1,547 +1,598 @@
+			require.NoError(err)
 
-		daemonTestWithPaymentChannel(t, &payer, &target, amt, eol, func(d *th.TestDaemon, channelID *types.ChannelID) {
-			ls := listChannelsAsStrs(d, &payer)[0]
+			assert.Len(channels, 1)
 
-			assert.Equal(fmt.Sprintf("%s: target: %s, amt: 10000, amt redeemed: 0, eol: 20", channelID, target.String()), ls)
+			channel := channels[chanid.String()]
+			assert.Equal(channelAmount, channel.Amount)
+			assert.Equal(channelExpiry, channel.Eol)
+			assert.Equal(targetAddr, channel.Target)
+			assert.Equal(types.ZeroAttoFIL, channel.AmountRedeemed)
 		})
 	})
 
 	t.Run("Works with specified payer", func(t *testing.T) {
 		t.Parallel()
+		require := require.New(t)
+		assert := assert.New(t)
+
+		// This test should run in 20 block times
+		ctx, cancel := context.WithDeadline(context.Background(), time.Now().Add(20*series.GlobalSleepDelay))
+		defer cancel()
+
+		// Get basic testing environment
+		env, _, NewNode, ctx := fasting.BasicFastSetup(ctx, t, fast.EnvironmentOpts{})
+
+		// Teardown after test ends
+		defer func() {
+			err := env.Teardown(ctx)
+			require.NoError(err)
+		}()
 
-		payer, err := address.NewFromString(fixtures.TestAddresses[2])
+		// Start test
+		targetDaemon := NewNode()
+		payerDaemon := NewNode()
+
+		addrs, err := targetDaemon.AddressLs(ctx)
 		require.NoError(err)
-		target, err := address.NewFromString(fixtures.TestAddresses[1])
+		targetAddr := addrs[0]
+
+		addrs, err = payerDaemon.AddressLs(ctx)
 		require.NoError(err)
+		payerAddr := addrs[0]
 
-		eol := types.NewBlockHeight(20)
-		amt := types.NewAttoFILFromFIL(10000)
+		channelExpiry := types.NewBlockHeight(20)
+		channelAmount := types.NewAttoFILFromFIL(1000)
 
-		daemonTestWithPaymentChannel(t, &payer, &target, amt, eol, func(d *th.TestDaemon, channelID *types.ChannelID) {
-			args := []string{"paych", "ls"}
-			args = append(args, "--from", target.String())
-			args = append(args, "--payer", payer.String())
+		WithPaymentChannel(ctx, t, payerDaemon, payerAddr, targetAddr, channelAmount, channelExpiry, func(chanid *types.ChannelID) {
+			channels, err := payerDaemon.PaychLs(ctx, fast.AOPayer(payerAddr))
+			require.NoError(err)
 
-			ls := th.RunSuccessLines(d, args...)[0]
+			assert.Len(channels, 1)
 
-			assert.Equal(fmt.Sprintf("%s: target: %s, amt: 10000, amt redeemed: 0, eol: 20", channelID, target.String()), ls)
+			channel := channels[chanid.String()]
+			assert.Equal(channelAmount, channel.Amount)
+			assert.Equal(channelExpiry, channel.Eol)
+			assert.Equal(targetAddr, channel.Target)
+			assert.Equal(types.ZeroAttoFIL, channel.AmountRedeemed)
 		})
 	})
 
-	t.Run("Notifies when channels not found", func(t *testing.T) {
+	t.Run("No results when listing with different from address", func(t *testing.T) {
 		t.Parallel()
+		require := require.New(t)
+		assert := assert.New(t)
+
+		// This test should run in 20 block times
+		ctx, cancel := context.WithDeadline(context.Background(), time.Now().Add(20*series.GlobalSleepDelay))
```

Comment:
Seeing this pattern is used in every payment channel test, should we consider making it a private helper method?
```go
func fastSetupWithBlockDeadline(t *testing.T, deadline time.Duration) (context.Context, fast.Environment, func() *fast.Filecoin) 
```